### ISSUE TITLE
Errors sent in response to restartFrame are not shown to user

### ISSUE THREAD

**Issue by DanTup** [microsoft/vscode#87605]
In `restartFrameRequest` I catch errors from my debugger and send back error responses like this:

```ts
protected restartFrameRequest(response: DebugProtocol.RestartFrameResponse, args: DebugProtocol.RestartFrameArguments): void {
	// snip ...

	thread.resume("Rewind", frame.index).then((_) => {
		this.sendResponse(response);
	}).catch((error) => {
		response.success = false;
		response.message = `Failed to restart frame ${error}`;
		this.sendResponse(response);
	});
}
```

However the errors sent back with `success=false` are not shown anywhere to the user so it's not clear why nothing is happening when `restartFrame` is not valid.

**Comment by isidorn** [comment]
This was a general issue on the VS Code side that we are not preoprly respecting the `success` field in the response.
I have changed this, that we respect the success = false the same way if the adapter returned an error. 

Also the restartFrame command will know use the notifications to show an error if it happens.

@DanTup you can try it out when we release a new insiders and let me know if the issue is fixed for you. Though not sure when there will be a new insiders due to holidays. Thanks

**Comment by isidorn** [comment]
I take my previous comment back: there was no general issue on the vscode side. We were properly respecting the `success` field, I was just wrongly reading the code.
However I have still pushed that the restartFrame shows errors to the user thus this issue should be fixed.

**Comment by DanTup** [comment]
Thanks! Looks like Insiders hasn't been updated over the holiday, but I'll test this once it updates.

**Comment by isidorn** [comment]
It should be updated on Monday. Thanks

**Comment by connor4312** [comment]
In a debug adapter, I returned this response:

```json
{
  "seq": 154,
  "type": "response",
  "request_seq": 35,
  "command": "restartFrame",
  "success": false,
  "message": "Cannot restart asynchronous frame",
  "body": {
    "error": {
      "id": 9223,
      "format": "Cannot restart asynchronous frame",
      "showUser": true
    }
  }
}
```

however I'm not seeing any error notification. This can be repro'd by trying to restart an async stack frame using the new js-debug, for instance

```js
setTimeout(() => {
  debugger; <- from here, restart the frame before the <Timeout>
}, 1);
```

**Comment by weinand** [comment]
@isidorn please take a look...

**Comment by isidorn** [comment]
Nothing for this milestone, moving to February.

**Comment by isidorn** [comment]
This should be fixed via the fc6e8fdac2e6a4ffce304ec7752bc028ddae289f

**Comment by connor4312** [comment]
Awesome, thanks!
